hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableTokenStoreImpl.java

```diff
@@ -58,6 +59,13 @@ public TokenMetadata getTokenMeta(@NonNull final TokenID id) {
         return tokenMetaFrom(token.get());
     }
 
+    @Override
+    @Nullable
+    public Token get(@NonNull final TokenID id) {
+        requireNonNull(id);
+        return getTokenLeaf(id.tokenNum()).orElse(null);
```

Comment:
We should not be doing `orElse(null)` imo. We could check `isPresent()` in handler.

Reply:
Can you explain? I don't like passing around `Optional` objects, they're super annoying having to `.get()` anywhere. I don't see why `orElse(null)` is bad?